conjure-java-core/src/main/java/com/palantir/conjure/java/services/Retrofit2ServiceGenerator.java

```diff
@@ -280,4 +401,69 @@ private static ClassName httpMethodToClassName(String method) {
         throw new IllegalArgumentException("Unrecognized HTTP method: " + method);
     }
 
+    /**
+     * Indicates whether a particular type has a defaultable value.
+     */
+    private static final Type.Visitor<Boolean> TYPE_DEFAULTABLE_PREDICATE = new DefaultTypeVisitor<Boolean>() {
+        @Override
+        public Boolean visitOptional(OptionalType value) {
+            return true;
+        }
+
+        @Override
+        public Boolean visitList(ListType value) {
+            return true;
+        }
+
+        @Override
+        public Boolean visitSet(SetType value) {
+            return true;
+        }
+
+        @Override
+        public Boolean visitMap(MapType value) {
+            return true;
+        }
+
+        @Override
+        public Boolean visitDefault() {
+            return false;
+        }
+    };
```

Comment:
I don't really want two copies of an identical TYPE_DEFAULTABLE_PREDICATE if poss - separate private class?